Tasks/FtpUpload/ftpuploadtask.ts

```diff
@@ -0,0 +1,243 @@
+/// <reference path="../../definitions/vsts-task-lib.d.ts" />
+/// <reference path="../../definitions/node.d.ts" />
+/// <reference path="../../definitions/Q.d.ts" />
+
+import path = require('path');
+import tl = require('vsts-task-lib/task');
+import url = require('url');
+import Q = require('q');
+
+import ftputils = require('./ftputils');
+
+var SortedSet = require('collections/sorted-set');
+var Client = require('ftp');
+
+var win = tl.osType().match(/^Win/);
+tl.debug('win: ' + win);
+
+var repoRoot: string = tl.getVariable('build.sourcesDirectory');
+function makeAbsolute(normalizedPath: string): string {
+    tl.debug('makeAbsolute:' + normalizedPath);
```

Comment:
I forget the exact reason, but I think it might have been you specified something absolute on the build machine, e.g. c:\temp